dist/prototypes.py

```diff
@@ -4,55 +4,80 @@
 
-        # Ignore statics in XXX.c files.
-        if fnmatch.fnmatch(name, "*.c") and 'static' in l:
-                continue
+    # If there's no CPP syntax, join everything.
+    if not '#endif' in ret:
+        ret = " ".join(ret.split())
 
-        # Join the first two lines, type and function name.
-        l = l.replace("\n", " ", 1)
+    # If it's not an inline function, prefix with "extern".
+    if not 'inline' in ret:
+        ret = 'extern ' + ret
 
-        # If there's no CPP syntax, join everything.
-        if not '#endif' in l:
-            l = " ".join(l.split())
+    # Switch to the include file version of any gcc attributes.
+    ret = ret.replace("WT_GCC_FUNC_ATTRIBUTE", "WT_GCC_FUNC_DECL_ATTRIBUTE")
 
-        # If it's not an inline function, prefix with "extern".
-        if not 'inline' in l:
-                l = 'extern ' + l
+    # Everything but void requires using any return value.
+    if not re.match(r'(static inline|extern) void', ret):
+        ret = ret + " WT_GCC_FUNC_DECL_ATTRIBUTE((warn_unused_result))"
 
-        # Switch to the include file version of any gcc attributes.
-        l = l.replace("WT_GCC_FUNC_ATTRIBUTE", "WT_GCC_FUNC_DECL_ATTRIBUTE")
+    # If a line ends in #endif, appending a semicolon results in an illegal
+    # expression, force an appended newline.
+    if re.match(r'#endif$', ret):
+        ret = ret + '\n'
 
-        # Everything but void requires using any return value.
-        if not re.match(r'(static inline|extern) void', l):
-                l = l + " WT_GCC_FUNC_DECL_ATTRIBUTE((warn_unused_result))"
+    return ret + ';\n'
 
-        # If a line ends in #endif, appending a semicolon results in an illegal
-        # expression, force an appended newline.
-        if re.match(r'#endif$', l):
-            l = l + '\n'
+# Find function prototypes in a file matching a given regex. Cleans the
+# function names to the point being immediately usable.
+def extract_prototypes(filename, regexp):
+    ret = []
+    s = open(filename, 'r').read()
+    for p in re.findall(regexp, s):
+        clean = clean_function_name(filename, p)
+        if clean is not None:
+            ret.append(clean)
 
-        # Add the trailing semi-colon.
-        list.append(l + ';\n')
+    return ret
+
+# Build function prototypes from a list of files.
+def fn_prototypes(fns, tests, name):
+    for sig in extract_prototypes(name, r'\n[A-Za-z_].*\n__wt_[^{]*'):
+        fns.append(sig)
+
+    for sig in extract_prototypes(name, r'\n[A-Za-z_].*\n__ut_[^{]*'):
+        tests.append(sig)
 
 # Write results and compare to the current file.
-def output(p, f):
+# Unit-testing functions are exposed separately in their own section to
+# allow them to be ifdef'd out.
+def output(fns, tests, f):
     tmp_file = '__tmp'
     tfile = open(tmp_file, 'w')
-    for e in sorted(list(set(p))):
+    for e in sorted(list(set(fns))):
+        tfile.write(e)
+
+    tfile.write('\n#ifdef HAVE_UNITTEST\n')
```

Comment:
This seems related to Etienne's comment [here](https://github.com/wiredtiger/wiredtiger/pull/7519/files?diff=split&w=0#r816493665). 
Can we change this to:
```
if len(tests) > 0:
    tfile.write('\n#ifdef HAVE_UNITTEST\n')
    ...
```

and avoid the empty `#ifdef`s in `extern_win.h`/`extern_posix.h`?

Reply:
I suggest we leave this optimisation to be considered later.